packages/discussions/src/utils/channels/can-post-to-channel.ts

```diff
@@ -99,15 +99,19 @@ function channelAllowsPostsByThisUsersGroups(
     const {
       id: userGroupId,
       userMembership: { memberType: userMemberType },
+      typeKeywords,
     } = userGroup;
 
     const channelGroupPermission = acl.groups[userGroupId];
 
-    if (!channelGroupPermission) {
+    if (
+      !channelGroupPermission ||
+      !ALLOWED_GROUP_ROLES.includes(userMemberType) ||
+      typeKeywords.includes("cannotDiscuss")
```

Comment:
Optional, maybe make `cannotDiscuss` a `const` and reference here and below, not sure if we'll be using this typekeyword value in the future for content.

Could just return the result of the expression if we invert its conditions.

```ts
return (
  channelGroupPermission &&
  ALLOWED_GROUP_ROLES.includes(userMemberType) &&
  !typeKeywords.includes("cannotDiscuss")
);
```

Reply:
Changed in https://github.com/Esri/hub.js/pull/892/commits/5b262ef91e8eaa66b798c2a6c5a28fbb2368754c.